src/helpers/privacy_budget.rs

```diff
@@ -0,0 +1,217 @@
+use crate::error::{Error, Res};
+use crate::threshold::Ciphertext;
+use hex::encode as hex;
+/// This module implements privacy budget using a redis cache
+use redis::{transaction, Commands, RedisError};
+use rust_elgamal::{DecryptionKey, RistrettoPoint, Scalar};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+struct BlindedMatchKey(Ciphertext);
+
+#[allow(dead_code)]
+impl BlindedMatchKey {
+    pub fn new(key: DecryptionKey, message: RistrettoPoint, scalar: Scalar) -> Self {
+        let c = key.encryption_key().encrypt_with(message, scalar);
+        Self(c)
+    }
+}
+
+impl ToString for BlindedMatchKey {
```

Comment:
I think that you want to implement ToRedisArgs instead.  Then you don't need hex encoding.